### ISSUE TITLE
Random segfault

### ISSUE THREAD

**Issue by romainreuillon** [indigo-dc/udocker#102]
I get random segfault when running:
```udocker run ubuntu:latest bash -c "apt update ; apt install -y cowsay"```

Around 1 on 2 times I get:

```
****************************************************************************** 
 *                                                                            * 
 *               STARTING 852019b1-51db-34c5-8de2-84deead509bd                * 
 *                                                                            * 
 ****************************************************************************** 
 executing: bash
Get:1 http://security.ubuntu.com/ubuntu xenial-security InRelease [102 kB]
Hit:2 http://archive.ubuntu.com/ubuntu xenial InRelease  
Get:3 http://archive.ubuntu.com/ubuntu xenial-updates InRelease [102 kB]           
Get:4 http://archive.ubuntu.com/ubuntu xenial-backports InRelease [102 kB]                  
Get:5 http://security.ubuntu.com/ubuntu xenial-security/universe Sources [53.1 kB]
Get:6 http://archive.ubuntu.com/ubuntu xenial/universe Sources [9802 kB]        
Get:7 http://security.ubuntu.com/ubuntu xenial-security/main amd64 Packages [505 kB]                  
Get:8 http://security.ubuntu.com/ubuntu xenial-security/universe amd64 Packages [229 kB] 
Get:9 http://security.ubuntu.com/ubuntu xenial-security/multiverse amd64 Packages [3479 B]
Get:10 http://archive.ubuntu.com/ubuntu xenial/universe amd64 Packages [9827 kB]
Get:11 http://archive.ubuntu.com/ubuntu xenial/multiverse amd64 Packages [176 kB]                                                                                      
Get:12 http://archive.ubuntu.com/ubuntu xenial-updates/universe Sources [231 kB]                                                                                       
Get:13 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 Packages [866 kB]                                                                                    
Get:14 http://archive.ubuntu.com/ubuntu xenial-updates/restricted amd64 Packages [13.7 kB]                                                                             
Get:15 http://archive.ubuntu.com/ubuntu xenial-updates/universe amd64 Packages [719 kB]                                                                                
Get:16 http://archive.ubuntu.com/ubuntu xenial-updates/multiverse amd64 Packages [18.5 kB]                                                                             
Get:17 http://archive.ubuntu.com/ubuntu xenial-backports/main amd64 Packages [5174 B]                                                                                  
Get:18 http://archive.ubuntu.com/ubuntu xenial-backports/universe amd64 Packages [7150 B]                                                                              
Fetched 22.8 MB in 7s (2873 kB/s)                                                                                                                                      
Reading package lists... Done
Building dependency tree       
Reading state information... Done
2 packages can be upgraded. Run 'apt list --upgradable' to see them.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  cowsay-off ifupdown iproute2 isc-dhcp-client isc-dhcp-common libatm1 libdns-export162 libgdbm3 libisc-export160 libmnl0 libperl5.22 libtext-charwidth-perl
  libxtables11 netbase perl perl-modules-5.22 rename
Suggested packages:
  filters ppp rdnssd iproute2-doc resolvconf avahi-autoipd isc-dhcp-client-ddns apparmor perl-doc libterm-readline-gnu-perl | libterm-readline-perl-perl make
The following NEW packages will be installed:
  cowsay cowsay-off ifupdown iproute2 isc-dhcp-client isc-dhcp-common libatm1 libdns-export162 libgdbm3 libisc-export160 libmnl0 libperl5.22 libtext-charwidth-perl
  libxtables11 netbase perl perl-modules-5.22 rename
0 upgraded, 18 newly installed, 0 to remove and 2 not upgraded.
Need to get 8149 kB of archives.
After this operation, 44.8 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu xenial/main amd64 libatm1 amd64 1:2.5.1-1.5 [24.2 kB]
Get:2 http://archive.ubuntu.com/ubuntu xenial/main amd64 libmnl0 amd64 1.0.3-5 [12.0 kB]
Get:3 http://archive.ubuntu.com/ubuntu xenial/main amd64 libgdbm3 amd64 1.8.3-13.1 [16.9 kB]
Get:4 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 perl-modules-5.22 all 5.22.1-9ubuntu0.2 [2661 kB]
Get:5 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libperl5.22 amd64 5.22.1-9ubuntu0.2 [3391 kB]
Get:6 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 perl amd64 5.22.1-9ubuntu0.2 [237 kB]
Get:7 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 iproute2 amd64 4.3.0-1ubuntu3.16.04.2 [522 kB]
Get:8 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 ifupdown amd64 0.8.10ubuntu1.2 [54.9 kB]
Get:9 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libisc-export160 amd64 1:9.10.3.dfsg.P4-8ubuntu1.9 [153 kB]
Get:10 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libdns-export162 amd64 1:9.10.3.dfsg.P4-8ubuntu1.9 [666 kB]
Get:11 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 isc-dhcp-client amd64 4.3.3-5ubuntu12.7 [223 kB]
Get:12 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 isc-dhcp-common amd64 4.3.3-5ubuntu12.7 [105 kB]
Get:13 http://archive.ubuntu.com/ubuntu xenial/main amd64 libtext-charwidth-perl amd64 0.04-7build5 [9248 B]
Get:14 http://archive.ubuntu.com/ubuntu xenial/main amd64 libxtables11 amd64 1.6.0-2ubuntu3 [27.2 kB]
Get:15 http://archive.ubuntu.com/ubuntu xenial/main amd64 netbase all 5.3 [12.9 kB]
Get:16 http://archive.ubuntu.com/ubuntu xenial/universe amd64 cowsay all 3.03+dfsg1-15 [18.0 kB]
Get:17 http://archive.ubuntu.com/ubuntu xenial/universe amd64 cowsay-off all 3.03+dfsg1-15 [3640 B]
Get:18 http://archive.ubuntu.com/ubuntu xenial/main amd64 rename all 0.20-4 [12.0 kB]
Fetched 8149 kB in 2s (2908 kB/s) 
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package libatm1:amd64.
(Reading database ... 4768 files and directories currently installed.)
Preparing to unpack .../libatm1_1%3a2.5.1-1.5_amd64.deb ...
Unpacking libatm1:amd64 (1:2.5.1-1.5) ...
Selecting previously unselected package libmnl0:amd64.
Preparing to unpack .../libmnl0_1.0.3-5_amd64.deb ...
Unpacking libmnl0:amd64 (1.0.3-5) ...
Selecting previously unselected package libgdbm3:amd64.
Preparing to unpack .../libgdbm3_1.8.3-13.1_amd64.deb ...
Unpacking libgdbm3:amd64 (1.8.3-13.1) ...
Selecting previously unselected package perl-modules-5.22.
Preparing to unpack .../perl-modules-5.22_5.22.1-9ubuntu0.2_all.deb ...
Unpacking perl-modules-5.22 (5.22.1-9ubuntu0.2) ...
Selecting previously unselected package libperl5.22:amd64.
Preparing to unpack .../libperl5.22_5.22.1-9ubuntu0.2_amd64.deb ...
Unpacking libperl5.22:amd64 (5.22.1-9ubuntu0.2) ...
Selecting previously unselected package perl.
Preparing to unpack .../perl_5.22.1-9ubuntu0.2_amd64.deb ...
Unpacking perl (5.22.1-9ubuntu0.2) ...
Selecting previously unselected package iproute2.
Preparing to unpack .../iproute2_4.3.0-1ubuntu3.16.04.2_amd64.deb ...
Unpacking iproute2 (4.3.0-1ubuntu3.16.04.2) ...
Selecting previously unselected package ifupdown.
Preparing to unpack .../ifupdown_0.8.10ubuntu1.2_amd64.deb ...
Unpacking ifupdown (0.8.10ubuntu1.2) ...
Selecting previously unselected package libisc-export160.
Preparing to unpack .../libisc-export160_1%3a9.10.3.dfsg.P4-8ubuntu1.9_amd64.deb ...
Unpacking libisc-export160 (1:9.10.3.dfsg.P4-8ubuntu1.9) ...
Selecting previously unselected package libdns-export162.
Preparing to unpack .../libdns-export162_1%3a9.10.3.dfsg.P4-8ubuntu1.9_amd64.deb ...
Unpacking libdns-export162 (1:9.10.3.dfsg.P4-8ubuntu1.9) ...
Selecting previously unselected package isc-dhcp-client.
Preparing to unpack .../isc-dhcp-client_4.3.3-5ubuntu12.7_amd64.deb ...
Unpacking isc-dhcp-client (4.3.3-5ubuntu12.7) ...
Selecting previously unselected package isc-dhcp-common.
Preparing to unpack .../isc-dhcp-common_4.3.3-5ubuntu12.7_amd64.deb ...
Unpacking isc-dhcp-common (4.3.3-5ubuntu12.7) ...
Selecting previously unselected package libtext-charwidth-perl.
Preparing to unpack .../libtext-charwidth-perl_0.04-7build5_amd64.deb ...
Unpacking libtext-charwidth-perl (0.04-7build5) ...
Selecting previously unselected package libxtables11:amd64.
Preparing to unpack .../libxtables11_1.6.0-2ubuntu3_amd64.deb ...
Unpacking libxtables11:amd64 (1.6.0-2ubuntu3) ...
Selecting previously unselected package netbase.
Preparing to unpack .../archives/netbase_5.3_all.deb ...
Unpacking netbase (5.3) ...
Selecting previously unselected package cowsay.
Preparing to unpack .../cowsay_3.03+dfsg1-15_all.deb ...
Unpacking cowsay (3.03+dfsg1-15) ...
Selecting previously unselected package cowsay-off.
Preparing to unpack .../cowsay-off_3.03+dfsg1-15_all.deb ...
Unpacking cowsay-off (3.03+dfsg1-15) ...
dpkg-deb: error: subprocess <decompress> was killed by signal (Segmentation fault), core dumped
dpkg: error processing archive /var/cache/apt/archives/cowsay-off_3.03+dfsg1-15_all.deb (--unpack):
 subprocess dpkg-deb --fsys-tarfile returned error exit status 2
Selecting previously unselected package rename.
Preparing to unpack .../archives/rename_0.20-4_all.deb ...
Unpacking rename (0.20-4) ...
Processing triggers for systemd (229-4ubuntu21) ...
Processing triggers for libc-bin (2.23-0ubuntu9) ...
Errors were encountered while processing:
 /var/cache/apt/archives/cowsay-off_3.03+dfsg1-15_all.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

My system info is:
```Linux simplet 4.13.0-17-generic #20-Ubuntu SMP Mon Nov 6 10:04:08 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux```

And for udocker:
```
[reuillon:~/.openmole/simplet] 7m40s 1 $ udocker version
version: 1.1.1
tarball: https://owncloud.indigo-datacloud.eu/index.php/s/AFImjw8ii0X72xf/download https://cernbox.cern.ch/index.php/s/VC7GuVWA7mYRAiy/download http://repo.indigo-datacloud.eu/repository/indigo/2/centos7/x86_64/tgz/udocker-1.1.1.tar.gz
```

How I could help to diagnose this bug further?

cheers,
Romain

**Comment by romainreuillon** [comment]
It seem to work using the P2 mode. Damned seccomp !

**Comment by jorge-lip** [comment]
Hi Romain,
Indeed, I'm suspecting of spurious events when using SECCOMP, will check.
Jorge

**Comment by mariojmdavid** [comment]
I have run this 4 times in a row in 4 different machines, and did not occur a single time
considered solved in the latest pre release 1.2.9
if it happens again don't hesitate to re-open the issue
